front/lib/resources/storage/models/membership.ts

```diff
@@ -0,0 +1,77 @@
+import type { MembershipRoleType } from "@dust-tt/types";
+import type {
+  CreationOptional,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+} from "sequelize";
+import { DataTypes, Model } from "sequelize";
+
+import { User, Workspace } from "@app/lib/models";
+import { frontSequelize } from "@app/lib/resources/storage";
+
+export class MembershipModel extends Model<
+  InferAttributes<MembershipModel>,
+  InferCreationAttributes<MembershipModel>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare role: MembershipRoleType;
+  declare startAt: Date;
+  declare endAt: Date | null;
+
+  declare userId: ForeignKey<User["id"]>;
+  declare workspaceId: ForeignKey<Workspace["id"]>;
+}
+MembershipModel.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    role: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    startAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+    endAt: {
+      type: DataTypes.DATE,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "membership",
+    sequelize: frontSequelize,
+    indexes: [
+      { fields: ["userId", "role"] },
+      { fields: ["startAt"] },
+      { fields: ["endAt"] },
```

Comment:
From the queries we use, shouldn't this be `["endAt", "startAt"]` instead?

Reply:
You are right (although in practice it's almost the same).
Will do in separate PR since this one has no migration (just moved this mode)